tempesta_fw/http.c

```diff
@@ -2542,9 +2540,9 @@ tfw_http_resp_process(TfwConn *conn, struct sk_buff *skb, unsigned int off)
 bad_msg:
 	bad_req = tfw_http_popreq(hmresp);
 	if (bad_req)
-		tfw_http_conn_msg_free((TfwHttpMsg *)bad_req);
+		tfw_http_send_500(bad_req, "response dropped: processing error");
 	tfw_http_conn_msg_free(hmresp);
-	return r;
+	return TFW_PASS;
```

Comment:
I argue against returning `TFW_PASS` here. If we've reach `bad_msg` label, it means that we've encountered an error during parsing. This means that the errored response wasn't red fully from the connection. So then we would get the parse error for every next read from the connection.

If error code is returned, we will close the server connection from the Tempesta side. The `ss_tcp_state_change()` callback with state `TCP_CLOSE` will be called and we will re-establish connection to the server.

Reply:
Corrected.